File: objectModel/Java/objectmodel/src/main/java/com/microsoft/commondatamodel/objectmodel/resolvedmodel/expressionparser/InputValues.java
// Copyright (c) Microsoft Corporation. All rights reserved.
// Licensed under the MIT License. See License.txt in the project root for license information.

package com.microsoft.commondatamodel.objectmodel.resolvedmodel.expressionparser;

import com.microsoft.commondatamodel.objectmodel.resolvedmodel.projections.ProjectionDirective;

/**
 * A structure to carry all the input values during evaluation/resolution of an expression tree
 *
 * @deprecated This class is extremely likely to be removed in the public interface, and not
 * meant to be called externally at all. Please refrain from using it.
 */
@Deprecated
public class InputValues {
    private Integer nextDepth;
    private Integer maxDepth;
    private Boolean noMaxDepth;
    private Boolean isArray;

    private Integer minCardinality;
    private Integer maxCardinality;

    private Boolean referenceOnly;
    private Boolean normalized;
    private Boolean structured;
    private Boolean isVirtual;

    public InputValues() { }

    public InputValues(ProjectionDirective projDirective) {
        this.setNoMaxDepth(projDirective.getHasNoMaximumDepth());
        this.setIsArray(projDirective.getIsArray());
        this.setReferenceOnly(projDirective.getIsReferenceOnly());
        this.setNormalized(projDirective.getIsNormalized());
        this.setStructured(projDirective.getIsStructured());
        this.setIsVirtual(projDirective.getIsVirtual());
        this.setNextDepth(projDirective.getResOpt().depthInfo.getCurrentDepth());
        this.setMaxDepth(projDirective.getMaximumDepth());
        this.setMinCardinality(projDirective.getCardinality() != null ? projDirective.getCardinality().getMinimumNumber() : null);
        this.setMaxCardinality(projDirective.getCardinality() != null ? projDirective.getCardinality().getMaximumNumber() : null);
    }


    public Integer getNextDepth() {
        return nextDepth;
    }

    public void setNextDepth(final Integer nextDepth) {
        this.nextDepth = nextDepth;
    }

    public Integer getMaxDepth() {
        return maxDepth;
    }

    public void setMaxDepth(final Integer maxDepth) {
        this.maxDepth = maxDepth;
    }

    public Boolean getNoMaxDepth() {
        return noMaxDepth;
    }

    public void setNoMaxDepth(final Boolean noMaxDepth) {
        this.noMaxDepth = noMaxDepth;
    }

    public Boolean getIsArray() {
        return isArray;
    }

    public void setIsArray(final Boolean isArray) {
        this.isArray = isArray;
    }

    public Integer getMinCardinality() {
        return minCardinality;
    }

    public void setMinCardinality(final Integer minCardinality) {
        this.minCardinality = minCardinality;
    }

    public Integer getMaxCardinality() {
        return maxCardinality;
    }

    public void setMaxCardinality(final Integer maxCardinality) {
        this.maxCardinality = maxCardinality;
    }

    public Boolean getReferenceOnly() {
        return referenceOnly;
    }

    public void setReferenceOnly(final Boolean referenceOnly) {
        this.referenceOnly = referenceOnly;
    }

    public Boolean getNormalized() {
        return normalized;
    }

    public void setNormalized(final Boolean normalized) {
        this.normalized = normalized;
    }

    public Boolean getStructured() {
        return structured;
    }

    public void setStructured(final Boolean structured) {
        this.structured = structured;
    }

    public Boolean getIsVirtual() {
        return isVirtual;
    }

    public void setIsVirtual(final Boolean isVirtual) {
        this.isVirtual = isVirtual;
    }
}

File: objectModel/Python/cdm/resolvedmodel/expression_parser/input_values.py
# Copyright (c) Microsoft Corporation. All rights reserved.
# Licensed under the MIT License. See License.txt in the project root for license information.

from typing import Optional, TYPE_CHECKING

if TYPE_CHECKING:
    from cdm.resolvedmodel.projections.projection_directive import ProjectionDirective


class InputValues:
    """A structure to carry all the input values during evaluation/resolution of an expression tree"""

    def __init__(self, proj_directive: 'ProjectionDirective'):
        if not proj_directive:
            return

        self.no_max_depth = proj_directive._has_no_maximum_depth  # type: Optional[bool]
        self.is_array = proj_directive._is_array  # type: Optional[bool]

        self.reference_only = proj_directive._is_reference_only  # type: Optional[bool]
        self.normalized = proj_directive._is_normalized  # type: Optional[bool]
        self.structured = proj_directive._is_structured  # type: Optional[bool]
        self.is_virtual = proj_directive._is_virtual  # type: Optional[bool]

        self.next_depth = proj_directive._res_opt._depth_info.current_depth  # type: Optional[int]
        self.max_depth = proj_directive._maximum_depth  # type: Optional[int]
        
        self.min_cardinality = proj_directive._cardinality._minimum_number if proj_directive._cardinality else None  # type: Optional[int]
        self.max_cardinality = proj_directive._cardinality._maximum_number if proj_directive._cardinality else None  # type: Optional[int]
